Realm/RLMRealm.mm

```diff
@@ -257,6 +257,23 @@ void RLMSetAsyncOpenQueue(dispatch_queue_t queue) {
     return ret;
 }
 
++ (void)subscribeToInitialSubscriptionsFromRealm:(RLMRealm *)realm
+                                   configuration:(RLMRealmConfiguration *)configuration
+                                     isFirstOpen:(bool)isFirstOpen {
+    if (configuration.config.sync_config && configuration.syncConfiguration.enableFlexibleSync && configuration.initialSubscriptions) {
+#if REALM_ENABLE_SYNC
+        if (isFirstOpen || configuration.rerunOnOpen) {
+            RLMSyncSubscriptionSet *subscriptions = realm.subscriptions;
+            [subscriptions update:^{
+                configuration.initialSubscriptions(subscriptions);
+            }];
+        }
+#else
+    @throw RLMException(@"Realm was not compiled with sync enabled");
```

Comment:
this will throw every time a realm is opened without sync compiled, no need for the `#else`